src/backend/distributed/commands/truncate.c

```diff
@@ -395,81 +396,19 @@ LockTruncatedRelationMetadataInWorkers(TruncateStmt *truncateStatement)
 		List *referencingTableList = cacheEntry->referencingRelationsViaForeignKey;
 		foreach_oid(referencingRelationId, referencingTableList)
 		{
-			distributedRelationList = list_append_unique_oid(distributedRelationList,
-															 referencingRelationId);
+			referencingRelationIds = lappend_oid(referencingRelationIds,
+												 referencingRelationId);
 		}
 	}
 
-	if (distributedRelationList != NIL)
+	foreach_oid(relationId, referencingRelationIds)
```

Comment:
why don't we do list_append_unique_oid above and have another loop here?

Reply:
Also, we seem to have `list_member_oid ` check in the same loop. We should avoid one of these calls, whichever you think fits better

Reply:
I cleaned up how `distributedRelationList` is constructed. Let me know if you think that can be improved on.